src/main/java/reposense/authorship/FileInfoExtractor.java

```diff
@@ -109,20 +110,36 @@
                 continue;
             }
 
+            if (!isValidAndNonBinaryFile(filePath, nonBinaryFilesSet)) {
+                continue;
+            }
+
             if (Format.isInsideWhiteList(filePath, config.getFormats())) {
-                try {
-                    FileInfo currentFileInfo = generateFileInfo(config.getRepoRoot(), filePath);
-                    setLinesToTrack(currentFileInfo, fileDiffResult);
-                    fileInfos.add(currentFileInfo);
-                } catch (InvalidPathException ipe) {
-                    logger.warning(String.format(INVALID_FILE_PATH_MESSAGE_FORMAT, filePath));
-                }
+                FileInfo currentFileInfo = generateFileInfo(config.getRepoRoot(), filePath);
+                setLinesToTrack(currentFileInfo, fileDiffResult);
+                fileInfos.add(currentFileInfo);
             }
         }
 
         return fileInfos;
     }
 
+    /**
+     * Returns a {@code Set} of non-binary files for the repo {@code repoConfig}.
+     */
+    public static Set<Path> getListOfNonBinaryFiles(RepoConfiguration repoConfig) {
+        String rawGitDiffResult = GitDiff.gitGetModifiedFiles(Paths.get(repoConfig.getRepoRoot()));
```

Comment:
I think raw result is not very useful, it may be better to add another api in GitDiff e.g. gitGetModifiedFileList that provides the result as an ArrayList to reduce some clutter in FileInfoExtractor.

Reply:
Processed and returned the modified files as an ArrayList in the existing GitDiff method